test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -0,0 +1,516 @@
+    all_diff = tree.reserve_mapping(50, 25, rd2);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    VMATree::SingleDiff diff2 = all_diff.flag[NMTUtil::flag_to_index(mtNMT)];
+    EXPECT_EQ(-25, diff.reserve);
+    EXPECT_EQ(25, diff2.reserve);
+  }
+  { // Fully release reserved mapping
+    Tree::RegionData rd(NCS::StackIndex(), mtTest);
+    Tree tree;
+    VMATree::SummaryDiff all_diff = tree.reserve_mapping(0, 100, rd);
+    VMATree::SingleDiff diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(100, diff.reserve);
+    all_diff = tree.release_mapping(0, 100);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(-100, diff.reserve);
+  }
+  { // Convert some of a released mapping to a committed one
+    Tree::RegionData rd(NCS::StackIndex(), mtTest);
+    Tree tree;
+    VMATree::SummaryDiff all_diff = tree.reserve_mapping(0, 100, rd);
+    VMATree::SingleDiff diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(diff.reserve, 100);
+    all_diff = tree.commit_mapping(0, 100, rd);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(0, diff.reserve);
+    EXPECT_EQ(100, diff.commit);
+  }
+  { // Adjacent reserved mappings with same flag
+    Tree::RegionData rd(NCS::StackIndex(), mtTest);
+    Tree tree;
+    VMATree::SummaryDiff all_diff = tree.reserve_mapping(0, 100, rd);
+    VMATree::SingleDiff diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(diff.reserve, 100);
+    all_diff = tree.reserve_mapping(100, 100, rd);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(100, diff.reserve);
+  }
+  { // Adjacent reserved mappings with different flags
+  Tree::RegionData rd(NCS::StackIndex(), mtTest);
+    Tree::RegionData rd2(NCS::StackIndex(), mtNMT);
+    Tree tree;
+    VMATree::SummaryDiff all_diff = tree.reserve_mapping(0, 100, rd);
+    VMATree::SingleDiff diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(diff.reserve, 100);
+    all_diff = tree.reserve_mapping(100, 100, rd2);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtTest)];
+    EXPECT_EQ(0, diff.reserve);
+    diff = all_diff.flag[NMTUtil::flag_to_index(mtNMT)];
+    EXPECT_EQ(100, diff.reserve);
+  }
+
+  { // A commit with two previous commits inside of it should only register
+    // the new memory in the commit diff.
+    Tree tree;
+    Tree::RegionData rd(NCS::StackIndex(), mtTest);
+    tree.commit_mapping(128, 128, rd);
+    tree.commit_mapping(512, 128, rd);
+    VMATree::SummaryDiff diff = tree.commit_mapping(0, 1024, rd);
+    EXPECT_EQ(768, diff.flag[NMTUtil::flag_to_index(mtTest)].commit);
+    EXPECT_EQ(768, diff.flag[NMTUtil::flag_to_index(mtTest)].reserve);
+  }
+}
+
+// Exceedingly simple tracker for page-granular allocations
+// Use it for testing consistency with VMATree.
+struct SimpleVMATracker : public CHeapObj<mtTest> {
+  const size_t page_size = 4096;
+  enum Tpe { Reserved, Committed, Free };
```

Comment:
Wow, we save one letter! :-D

Reply:
Think about it, that's one whole byte per instance in the code! Very useful when pushing changes over a 75 baud connection to Mars :-P! Fixed.